backoffice/utils_proctorU_api.py

```diff
@@ -28,6 +29,27 @@
 BASE_URL = "https://" + settings.PROCTORU_API
 HEADER = {"Authorization-Token": settings.PROCTORU_TOKEN}
 
+def date_range(start_date, end_date, increment, period):
+    """
+    Split a date range in several time intervals
+    :param start_date: start of the interval
+    :param end_date: end of the interval
+    :param increment: value to increment (int)
+    :param period: type of "increment" (min, hour, day, month...)
+    :return: a list of couples with sub interval start and end [(start_date, int1), (int1, int2), ... , (int_n, end_date)]
+    """
+    result = []
+    nxt = start_date
+    delta = relativedelta.relativedelta(**{period:increment})
+    while nxt <= end_date:
+        result.append(nxt)
+        nxt += delta
+    result.append(end_date)
+
+    results = [(b, e) for b, e in zip(result[:-1], result[1:])]
+    return results
```

Comment:
A noter que ce code serait plus concis, plus rapide et nécessiterait moins de mémoire en utilisant des itérateurs : 

    nxt = start_date
    delta = relativedelta.relativedelta(**{period:increment})
    while nxt <= end_date:
        yield nxt, nxt + delta
        nxt += delta